### PR TITLE
Adding swagger for new SQL Pool rest APIs

### PR DESCRIPTION
**Pull request by tunafish88**
Description:
These APIs are being added for SQL Pools. Behind the scenes these APIs just make a call to the corresponding rest API for SQL Databases (datawarehouses in particular).
Currently for SQL databases dynamic datamaskingPolicy only Get, Put are Supported and not list. (This has always been the case and is by design). Similar explanation for 'SqlPoolVulnerabilityAssessmentRuleBaseline'.
SQL Pool is maintaining parity with whatever SQL DB offers.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [X] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ravbhatnagar** [comment]
As per Synapse RPs communication with SQL team - "Spoke to the owners of the SQL API and they say the original API is old and for historical reasons they have not supported the 'list' operation. Supporting it at this time is not an option , so I'm not sure if SQL RP doesn't support the operation, how we could support the operation from the Synapse RP, because we have nothing to call on the SQL RP side." 
"[2:21 PM] Shreya Roy Chowdhury
    The SQL Pool APIs (In Microsoft.Synapse RP) that we're writing, are basically proxying calls to Database APIs (in the Microsoft.Sql RP) 
    So we are documenting operations which have a corresponding operation on the Microsoft.SQL RP
    These 2 particular ones, do not have a list operation on the microsoft.sql rp ".
Signing off from ARM side as per above. Note that having list APIs is required as per RPC and is something every RP in azure has.  SIgning off to unblock. But ideally, you should work with SQL team to enable these APis.

**Comment by markcowl** [comment]
@tunafish88 Please fix the linting violations, as outlined here: https://aka.mc/ci-fix

**Comment by markcowl** [comment]
@tunafish88 You will need to follow the guidance here: http://aka.ms/ci-fix  to suppress the linter errors as the ARM team has signed off on them.

**Comment by tunafish88** [comment]
> @tunafish88 You will need to follow the guidance here: http://aka.ms/ci-fix to suppress the linter errors as the ARM team has signed off on them.

Thanks for pointing this out Mark. I actually did follow this, filled out a suppression request form, got the suppression approval, added the suppression in the readme file and sent a new iteration. I'm not sure why the error is still showing up. I reached out the arm reviewer (and ccd you) asking them about it.

**Review by markcowl — CHANGES_REQUESTED** [review]

File: specification/synapse/resource-manager/readme.md
Comment (markcowl) [inline specification/synapse/resource-manager/readme.md:76]: In this case, you'll need to follow the guidance like here: https://github.com/Azure/autorest/tree/97b68250afd96111f79047e24e22eeb82a30426f/src/autorest-core/test/variations/suppressions#example-limit-scope-to-the-one-document-namespace-yourapi-and-apply-the-filter-only-to-operationid-where-the-name-begins-with-legacy_

To add a suppression to only a specific path, first specify the from./where, then the action (suppress)

**Review by markcowl — CHANGES_REQUESTED** [review]

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:5766]: Mark, these APIs are proxying calls to SQL RP APIs. So we're keeping the SQL Pool swagger documentation in sync with the Sql DB swagger documentation. We understand that any changes here will cause a breaking change.

My understanding is that in late 2021/early 2022, the SQL team will come up with newer version of Sqlpool APIs and take ownership of them. Till then these APIs are unlikely to change.

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:5646]: Spoke to the current owner of the SQL API. The original SQL API is 6 years old and is unlikely to change. It is recommended that I stay consistent with the original SQL API's swagger where this value is ModelAsString: false.

**Review by markcowl — APPROVED** [review]

**Pull request merged**
